Network/Network.cpp

```diff
@@ -625,154 +654,277 @@ namespace WPEFramework
-                        }
-                        if (ip_address.s_addr == broadcast_addr1.s_addr)
-                        {
-                            LOGWARN("Interface and Broadcast IP is same, return false \n");
-                            response["supported"] = iarmData.isSupported;
-                            result = false;
-                            returnResponse(result)
-                        }
-                        if (gateway_address.s_addr == broadcast_addr2.s_addr)
-                        {
-                            LOGWARN("Gateway and Broadcast IP is same, return false \n");
-                            response["supported"] = iarmData.isSupported;
-                            result = false;
-                            returnResponse(result)
-                        }
+                        LOGWARN("Interface and Broadcast IP is same, return false \n");
+                        response["supported"] = iarmData.isSupported;
+                        result = false;
+                        returnResponse(result)
+                    }
+                    if (gateway_address.s_addr == broadcast_addr2.s_addr)
+                    {
+                        LOGWARN("Gateway and Broadcast IP is same, return false \n");
+                        response["supported"] = iarmData.isSupported;
+                        result = false;
+                        returnResponse(result)
                     }
                 }
-                if (IARM_RESULT_SUCCESS ==
-                    IARM_Bus_Call(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETSRVMGR_API_setIPSettings, (void *) &iarmData,
-                                  sizeof(iarmData)))
-                {
-                    response["supported"] = iarmData.isSupported;
-                    result = true;
-                }
-                else
-                    response["supported"] = iarmData.isSupported;
             }
+            else
+            {
+                response["supported"] = iarmData.isSupported;
+                result = true;
+                returnResponse(result)
+            }
+            if (IARM_RESULT_SUCCESS == IARM_Bus_Call(IARM_BUS_NM_SRV_MGR_NAME, IARM_BUS_NETSRVMGR_API_setIPSettings, (void *) &iarmData, sizeof(iarmData)))
+            {
+                response["supported"] = iarmData.isSupported;
+                result = true;
+            }
+            else
+                response["supported"] = iarmData.isSupported;
 
             returnResponse(result)
         }
 
-        uint32_t Network::getIPSettings(const JsonObject& parameters, JsonObject& response)
+
+	uint32_t Network::getIPSettings(const JsonObject& parameters, JsonObject& response)
         {
+            JsonObject internal;
+            JsonObject InternalResponse;
             bool result = false;
-            string interface = "";
+            string interface;
+            string ipversion;
+            internal ["interface"] = interface;
```

Comment:
You are passing **_un-inited_** variable as input to the internal function. Not using anything from what application requested at **_parameters_**